server/src/com/cloud/ha/HighAvailabilityManagerImpl.java

```diff
@@ -242,16 +242,19 @@ public void scheduleRestartForVmsOnHost(final HostVO host, boolean investigate)
 
         // send an email alert that the host is down
         StringBuilder sb = null;
+        List<VMInstanceVO> reorderedVMList = new ArrayList<VMInstanceVO>();
         if ((vms != null) && !vms.isEmpty()) {
             sb = new StringBuilder();
-            sb.append("  Starting HA on the following VMs: ");
+            sb.append("  Starting HA on the following VMs:");
             // collect list of vm names for the alert email
-            VMInstanceVO vm = vms.get(0);
-            if (vm.isHaEnabled()) {
-                sb.append(" " + vm);
-            }
-            for (int i = 1; i < vms.size(); i++) {
-                vm = vms.get(i);
+            for (int i = 0; i < vms.size(); i++) {
+                VMInstanceVO vm = vms.get(i);
+                String name = vm.getInstanceName();
+                if (name.startsWith("i-")) {
```

Comment:
@koushik-das To identify VRs in this list, vm.getType().equals(VirtualMachine.Type.DomainRouter) would be better, instead of relying on the Vm name. Also, in the mentioned code SSVM and CPVM get higher priority than UserVms.